### PR TITLE
5.2 Release

### PR DESCRIPTION
**Pull request by thedrow**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

## Description

This is the release PR for 5.2.
It'll contain all the release notes and final work required to release 5.2 GA.

### CODE REVIEW

File: docs/whatsnew-5.2.rst [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+you should come join us :ref:`on IRC <irc-channel>`
+or :ref:`our mailing-list <mailing-list>`.
+
+To read more about Celery you should go read the :ref:`introduction <intro>`.
+
+While this version is **mostly** backward compatible with previous versions
+it's important that you read the following section as this release
+is a new major version.
+
+This version is officially supported on CPython 3.6, 3.7 & 3.8 & 3.9
```
Comment (auvipy) [inline docs/whatsnew-5.2.rst:32]: python 3.6 dropped from CI no?

**Comment by matusvalo** [comment]
Is it possible to have #6942 included in 5.2 release? It should also help with memory leaks - see PR description for details.

File: docs/whatsnew-5.2.rst (hunk 1 above)
Comment (thedrow) [reply docs/whatsnew-5.2.rst:32]: Yup.

**Comment by auvipy** [comment]
> Is it possible to have #6942 included in 5.2 release? It should also help with memory leaks - see PR description for details.

this should be included

**Comment by nicolaerosia** [comment]
please consider https://github.com/celery/celery/pull/6957 , in my setup I have a lot of expired tasks during peak hours, this helps because the invalid tasks are dropped by rabbitmq

File: docs/whatsnew-5.2.rst [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+See :ref:`custom-task-cls-app-wide` for more details.
+
+Important Notes From 5.0
+------------------------
+
+Dropped support for Python 2.7 & 3.5
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Celery now requires Python 3.6 and above.
```
Comment (auvipy) [inline docs/whatsnew-5.2.rst:233]: 3.7?

Comment (thedrow) [reply docs/whatsnew-5.2.rst:233]: These are notes from 5.0 which we include for those who upgrade.
Should we avoid including those in this release?

Comment (auvipy) [reply docs/whatsnew-5.2.rst:233]: mmm in that case we can keep them

File: docs/whatsnew-5.2.rst [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+When replacing a task with another task, we now give an indication of the
+replacing nesting level through the ``replaced_task_nesting`` header.
+
+A task which was never replaced has a ``replaced_task_nesting`` value of 0.
+
+Kombu
+-----
+
+Starting from v5.2, the minimum required version is Kombu 5.2.0.
+
```
Comment (matusvalo) [inline docs/whatsnew-5.2.rst:220]: I miss information about automated killing of orphaned workers in prefork pool: https://github.com/celery/celery/pull/6942

Comment (matusvalo) [reply docs/whatsnew-5.2.rst:220]:
```suggestion

Prefork Workers Pool
---------------------

Now all orphaned worker processes are killed automatically when main process exits.

```

**Pull request merged**